src/select/SelectBaseState.ts

```diff
@@ -0,0 +1,176 @@
+import * as React from "react";
+import { SetState } from "reakit-utils/ts";
+import { CompositeStateReturn, CompositeState, CompositeActions } from "reakit";
+
+import { Item } from "./helpers/types";
+
+export function getIdFromValue(
+  valuesById: ValuesById,
+  selectedValue: string | null | undefined,
+) {
+  if (selectedValue == null) return null;
+
+  const selectedOption = valuesById.find(
+    option => option.value === selectedValue,
+  );
+  return selectedOption ? selectedOption.id : null;
+}
+
+function getValueFromId(
+  valuesById: ValuesById,
+  currentId: CompositeState["currentId"],
+) {
+  if (currentId == null) return;
+
+  const selectedOption = valuesById.find(option => option.id === currentId);
+  return selectedOption ? selectedOption.value : undefined;
+}
+
+export function useSelectBaseState<T extends CompositeStateReturn>(
```

Comment:
Is this taken from reakit codebase? any particular use case for using a generic for this?

Reply:
It's from Combobox.

I am not sure why it's needed, we can also type them directly right ?

Reply:
Yeah we can type them directly no need of generics. I think maybe they had 2d and 1d grid combobox that's why they used generic.

Reply:
Right, I removed the grid since it's not needed in this commit - https://github.com/timelessco/renderless-components/pull/147/commits/0637aea6608634f4ffe1adf446aad4751bff5d99